Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -2,35 +2,1345 @@
+#align zpow_mul zpow_mul
+
+@[to_additive mul_zsmul]
+theorem zpow_mul' (a : α) (m n : ℤ) : a ^ (m * n) = (a ^ n) ^ m := by rw [mul_comm, zpow_mul]
+#align zpow_mul' zpow_mul'
+
+section bit0
+
+set_option linter.deprecated false
+
+@[to_additive bit0_zsmul]
+theorem zpow_bit0 (a : α) : ∀ n : ℤ, a ^ bit0 n = a ^ n * a ^ n
+  | (n : ℕ) => by simp only [zpow_ofNat, ← Int.ofNat_bit0, pow_bit0]
+  | -[n+1] => by
+    simp [← mul_inv_rev, ← pow_bit0]
+    rw [negSucc_eq, bit0_neg, zpow_neg]
+    norm_cast
+#align zpow_bit0 zpow_bit0
+
+@[to_additive bit0_zsmul']
+theorem zpow_bit0' (a : α) (n : ℤ) : a ^ bit0 n = (a * a) ^ n :=
+  (zpow_bit0 a n).trans ((Commute.refl a).mul_zpow n).symm
+#align zpow_bit0' zpow_bit0'
+
+@[simp]
+theorem zpow_bit0_neg [HasDistribNeg α] (x : α) (n : ℤ) : (-x) ^ bit0 n = x ^ bit0 n := by
+  rw [zpow_bit0', zpow_bit0', neg_mul_neg]
+#align zpow_bit0_neg zpow_bit0_neg
+
+end bit0
+
+end DivisionMonoid
+
+section Group
+
+variable [Group G]
+
+@[to_additive add_one_zsmul]
+theorem zpow_add_one (a : G) : ∀ n : ℤ, a ^ (n + 1) = a ^ n * a
+  | (n : ℕ) => by simp only [← Int.ofNat_succ, zpow_ofNat, pow_succ']
+  | -[0+1] => by erw [zpow_zero, zpow_negSucc, pow_one, mul_left_inv]
+  | -[n + 1+1] => by
+    rw [zpow_negSucc, pow_succ, mul_inv_rev, inv_mul_cancel_right]
+    rw [Int.negSucc_eq, neg_add, add_assoc, neg_add_self, add_zero]
+    exact zpow_negSucc _ _
+#align zpow_add_one zpow_add_one
+
+@[to_additive zsmul_sub_one]
+theorem zpow_sub_one (a : G) (n : ℤ) : a ^ (n - 1) = a ^ n * a⁻¹ :=
+  calc
+    a ^ (n - 1) = a ^ (n - 1) * a * a⁻¹ := (mul_inv_cancel_right _ _).symm
+    _ = a ^ n * a⁻¹ := by rw [← zpow_add_one, sub_add_cancel]
+
+#align zpow_sub_one zpow_sub_one
+
+@[to_additive add_zsmul]
+theorem zpow_add (a : G) (m n : ℤ) : a ^ (m + n) = a ^ m * a ^ n := by
+  induction' n using Int.induction_on with n ihn n ihn
+  case hz => simp
+  · simp only [← add_assoc, zpow_add_one, ihn, mul_assoc]
+  · rw [zpow_sub_one, ← mul_assoc, ← ihn, ← zpow_sub_one, add_sub_assoc]
+#align zpow_add zpow_add
+
+@[to_additive add_zsmul_self]
+theorem mul_self_zpow (b : G) (m : ℤ) : b * b ^ m = b ^ (m + 1) := by
+  conv_lhs =>
+    congr
+    rw [← zpow_one b]
+  rw [← zpow_add, add_comm]
+#align mul_self_zpow mul_self_zpow
+
+@[to_additive add_self_zsmul]
+theorem mul_zpow_self (b : G) (m : ℤ) : b ^ m * b = b ^ (m + 1) := by
+  conv_lhs =>
+    congr
+    · skip
+    rw [← zpow_one b]
+  rw [← zpow_add, add_comm]
+#align mul_zpow_self mul_zpow_self
+
+@[to_additive sub_zsmul]
+theorem zpow_sub (a : G) (m n : ℤ) : a ^ (m - n) = a ^ m * (a ^ n)⁻¹ := by
+  rw [sub_eq_add_neg, zpow_add, zpow_neg]
+#align zpow_sub zpow_sub
+
+@[to_additive one_add_zsmul]
+theorem zpow_one_add (a : G) (i : ℤ) : a ^ (1 + i) = a * a ^ i := by rw [zpow_add, zpow_one]
+#align zpow_one_add zpow_one_add
+
+@[to_additive]
+theorem zpow_mul_comm (a : G) (i j : ℤ) : a ^ i * a ^ j = a ^ j * a ^ i :=
+  (Commute.refl _).zpow_zpow _ _
+#align zpow_mul_comm zpow_mul_comm
```

Comment:
All #align statements need to be duplicated for the to_additive versions. (This will be automated soon, but are still needed in this PR.)

Reply:
What are the names of the `to_additive` theorems that need to be aligned? Is it replacing `mul` by `add`?

Reply:
Done, thanks